rust/src/meas/bootstrap.json

```diff
@@ -158,6 +158,115 @@
+            },
+            {
+                "name": "bounds",
+                "c_type": "AnyObject *",
+                "rust_type": "OptionT",
+                "default": null,
+                "description": "Set bounds on the count to make the algorithm run in constant-time."
+            },
+            {
+                "name": "D",
+                "default": "AllDomain<int>",
+                "description": "Domain of the data type to be privatized. Valid values are VectorDomain<AllDomain<T>> or AllDomain<T>",
+                "is_type": true
+            },
+            {
+                "name": "QO",
+                "description": "Data type of the sensitivity, scale, and budget.",
+                "is_type": true
+            }
+        ],
+        "derived_types": [
+            {
+                "name": "T",
+                "rust_type": {
+                    "function": "get_atom",
+                    "params": [
+                        "D"
+                    ]
+                }
+            },
+            {
+                "name": "OptionT",
+                "rust_type": {
+                    "origin": "Option",
+                    "args": [
+                        {
+                            "origin": "Tuple",
+                            "args": ["T", "T"]
+                        }
+                    ]
+                },
+                "generics": ["T"]
+            }
+        ],
+        "ret": {
+            "c_type": "FfiResult<AnyMeasurement *>"
+        }
+    },
+    "make_base_discrete_laplace_cks20": {
+        "description": "Make a Measurement that adds noise from the discrete_laplace(`scale`) distribution to the input.\nUses the sampling algorithm from CKS20, The Discrete Gaussian for Differential Privacy.\nAdjust D to noise vector-valued data.",
+        "features": ["contrib"],
+        "args": [
+            {
+                "name": "scale",
+                "c_type": "void *",
+                "rust_type": "QO",
+                "description": "noise scale parameter for the distribution. `scale` == sqrt(2) * standard_deviation."
+            },
+            {
+                "name": "D",
+                "default": "AllDomain<int>",
+                "description": "Domain of the data type to be privatized. Valid values are VectorDomain<AllDomain<T>> or AllDomain<T>",
+                "is_type": true
+            },
+            {
+                "name": "QO",
+                "description": "Data type of the sensitivity, scale, and budget.",
+                "is_type": true
+            }
+        ],
+        "ret": {
+            "c_type": "FfiResult<AnyMeasurement *>"
+        }
+    },
+    "make_base_discrete_laplace": {
+        "description": "Make a Measurement that adds noise from the discrete_laplace(`scale`) distribution to the input.\nAdjust D to noise vector-valued data.",
```

Comment:
It'd be nice to explain how this chooses between linear and cks20. Unfortunately it's complicated by the mpfr situation. Do you think we could add a note without making it too confusing?

Reply:
I won't mention the mpfr complication here. MPFR only ever doesn't exist when the library is built in a very specific developer mode.